### ISSUE TITLE
audio not working for both incoming and outgoing

### ISSUE THREAD

**Issue by dillipdala** [bedefaced/sipdotnet#27]
when i tried to receive a call audio is not working for both incoming and outgoing call. And how to request a video call to from this application.

please help me on this.

**Comment by bedefaced** [comment]
Hi! Did you try [dev-branch](https://github.com/bedefaced/sipdotnet/tree/dev) of sipdotnet?

**Comment by FirsovMS** [comment]
Hi! I have some problem with audio. I use dev branch. When i use phone.PlaybackDevice() i receive list with Count is 0. I wrote my implementation with Marshalling and turned to the linphone_core_get_sound_devices method, but i have the same result :(
I work on Windows 7x64 with Visual Studio 2015 Community.
What doing i wrong?

**Comment by bedefaced** [comment]
@FirsovMS Did you try [Linphone client](http://linphone.org/releases/windows/Linphone-4.1.1-win32.exe)? Maybe it's #22 related. It seems that new liblinphone version (which the dev-branch uses) is buggy.

**Comment by dillipdala** [comment]
yes, i have tried from  dev-branch of sipdotnet. but still getting same issue.

**Comment by FirsovMS** [comment]
In Linphone Clinet it works perfect, and all devices defince correct. But in sipdotnet not. I try to use master branch, my soundcard define , call establish but microphone have no sound.

**Comment by FirsovMS** [comment]
I noticed that the problem is related to LinphoneCore's veracity. When I use LinphoneCore 3.11.2-158-g4c291e2 - media devices are determined, but my interlocutor does not hear me. In the debug console, I received the error message: Could not set QoS flow, error = 87.
With other versions of LinphoneCore devices are not defined at all.
![image](https://user-images.githubusercontent.com/24794526/38414092-99b9fc8e-399e-11e8-85d0-cf3b0870084f.png)
in the screenshot list of my devices from the device manager.

**Comment by FirsovMS** [comment]
Yes! I have the same problem. I think this problem exists on mediastreamer2 dll's. After a lot of attempts, I came to the conclusion that there are some problems with the wasapi interface implemented in this library. At least for me, on _LinphoneCore 3.11.2-158-g4c291e2_ everything works with dev-brunch, except for the microphone. 
I explicitly received a message in the debug console: 'Could not initialize the MSWASAPI audio input interface [80070057]'. 
The latest original linphone windows client has the same problem.

**Comment by dillipdala** [comment]
Hi 

anybody has solution?>

**Comment by LunevNF** [comment]
No solution yet? How to work if cannot talk with people?

**Comment by Ixreb** [comment]
First of all, thank you very much for this solution.

But, same problem here.

Everything works fine with master.
Everything works fine with linphone-client (multiple versions).
But with sipdotnet-dev it does not transfer outgoing or incoming audio.

I would be happy to use sipdotnet-master, but I need to pause calls and other cool functions of sipdotnet-dev.

GetPlaybackDevices() and GetCaptureDevices() return empty.

Any ideas?? I've tried everything, I've even tried different versions of Visual Studio.
